pkg/logs/epforwarder/epforwarder.go

```diff
@@ -0,0 +1,215 @@
+package epforwarder
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/auditor"
+	"github.com/DataDog/datadog-agent/pkg/logs/client"
+	"github.com/DataDog/datadog-agent/pkg/logs/client/http"
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	"github.com/DataDog/datadog-agent/pkg/logs/restart"
+	"github.com/DataDog/datadog-agent/pkg/logs/sender"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	eventTypeDBMSamples = "dbm-samples"
+	eventTypeDBMMetrics = "dbm-metrics"
+)
+
+var passthroughPipelineDescs = []passthroughPipelineDesc{
+	{
+		eventType:              eventTypeDBMSamples,
+		endpointsConfigPrefix:  "database_monitoring.samples",
+		hostnameEndpointPrefix: "dbquery-http-intake.logs.",
+	},
+	{
+		eventType:              eventTypeDBMMetrics,
+		endpointsConfigPrefix:  "database_monitoring.metrics",
+		hostnameEndpointPrefix: "dbmetrics-http-intake.logs.",
+	},
+}
+
+// An EventPlatformForwarder forwards Messages to a destination based on their event type
+type EventPlatformForwarder interface {
+	SendEventPlatformEvent(e *message.Message, eventType string) error
+	Purge() map[string][]*message.Message
+	Start()
+	Stop()
+}
+
+type defaultEventPlatformForwarder struct {
+	mx              sync.RWMutex
+	pipelines       map[string]*passthroughPipeline
+	destinationsCtx *client.DestinationsContext
+}
+
+func (s *defaultEventPlatformForwarder) SendEventPlatformEvent(e *message.Message, eventType string) error {
+	s.mx.RLock()
+	p, ok := s.pipelines[eventType]
```

Comment:
Just saw https://github.com/DataDog/datadog-agent/commit/84cc39e026a3e8b41d8b3916577405125779dce8

But I'm wondering why do we need to protect `s.pipelines` with a mutex ?

It seems not changed after being created in `newDefaultEventPlatformForwarder`.

Reply:
Good point, there is no reason to protect it. Removed the read lock.

We need only one lock on `Purge` to ensure that only one purge can happen at a time.

Reply:
@djova can't that create a read map access if `SendEventPlatformEvent` is called at the same time as `Purge`?

Reply:
Yeah concurrent reads from a map are fine as long as no one is writing to it, which is true in this case, since the map doesn't change after creation. 

During a `Purge`, yes another thread might call `SendEventPlatformEvent`, in which case it could put objects on a channel as it's being purged, which should still be fine, as the purge will almost certainly reach the end of the channel faster than it's being written to.